src/scenes/NftDetailPage/NftDetailPage.tsx

```diff
@@ -28,6 +29,9 @@ function NftDetailPage({ nftId }: Props) {
   const isFromCollectionPage =
     globalThis?.sessionStorage?.getItem('prevPage') === `/${username}/${collectionId}`;
 
+  const authenticatedUsername = useAuthenticatedUsername();
+  const userOwnsAsset = authenticatedUsername === username;
```

Comment:
ah so this isn't documented well, but unfortunately `useAuthenticatedUsername` will throw an error if you're not logged in. this is bc it calls `useAuthenticatedUser` under the hood. i tried going to an NFT detail page while logged out (in incognito), and was indeed able to trigger the error.

![image](https://user-images.githubusercontent.com/12162433/153130642-df213164-4cf2-4ed4-8b83-6142b2fb35d2.png)

there is a more lax hook you can use from the same file: `usePossiblyAuthenticatedUser`. so you can update like this:

```js
  const authenticatedUser = usePossiblyAuthenticatedUser(); // returns a user object OR or `undefined` if not logged in
  const userOwnsAsset = authenticatedUser?.username === username;
```